docs/arch/adr-009-storage-migration.md

```diff
@@ -0,0 +1,94 @@
+# ADR 9: OpenEBS to OpenEBS + Rook Migration
+
+## Context
+
+Single node installs should not use Rook, as doing so adds complexity, increases resource usage, and invites disk corruption.
+Multi node installs very often require distributed storage, best provided by Rook.
+
+Many customers will move from a single-node install to a multi-node install as circumstances change.
+This migration should be as simple as possible.
+Vendors generally should also not need to specify multiple installer specs for single and multi-node installs.
+
+## Decision
+
+Provide a migration path from OpenEBS to OpenEBS + Rook, triggered by the presence of 3+ nodes.
+
+## Solution
+
+A new field, "minimum node count", will be added to the rook config. (minimum value 2, requires openebs localpv to be present)
+When it is set, the rook operator will be installed on single node installs, but no ceph cluster will be created.
+
+An example spec is provided below:
+
+```yaml
+spec:
+  rook:
+    version: "1.10.x"
+    storageClassName: "distributed"
+    minimumNodeCount: 3
+  openebs:
+    version: "3.4.x"
+    isLocalPVEnabled: true
+    localPVStorageClassName: "local"
+```
+
+When installed on a single node, two storageclasses will be available - "scaling" and "local". 
```

Comment:
Calling it _scaling_ doesn't seem to convey it's purpose clear enough for me. Perhaps we can call it _centralized_ which is the opposite of _distributed_? _Centralized_ makes sense to me as it convey cluster storage that is currently designated for this node aside from calling it something like 'local 2`/ 'second local`.

Reply:
how about "dynamic"?

Reply:
"elastic"?

Reply:
'dynamic' would be fine with me

Reply:
I'm going to merge this now though, and we can make changes to it afterwards

Reply:
`dynamic` is better but I still think looking at the SC list, after a third node is added, that it better conveys the state of the cluster (i.e. that we went from non-distributed to distributed storage).
```
kubectl get sc
NAME                        PROVISIONER                                                RECLAIMPOLICY   VOLUMEBINDINGMODE      ALLOWVOLUMEEXPANSION   AGE
local                       openebs.io/local                                           Delete          WaitForFirstConsumer   false                  16d
centralized                 openebs.io/local                                           Delete          WaitForFirstConsumer   false                  16d
distributed (default)       rook-ceph.rbd.csi.ceph.com                                          
```